racket/collects/racket/syntax.rkt

```diff
@@ -113,6 +113,8 @@
                    #:subs? [subs? #f]
                    #:subs-intro [subs-intro (default-intro)]
                    fmt . args)
+  (unless (or (syntax? lctx) (eq? lctx #f))
+    (raise-argument-error 'format-id "(or/c syntax? #f)" lctx))
```

Comment:
I think `(apply raise-argument-error 'format-id "(or/c syntax? #f)" 0 lctx fmt args)` might make it easier to locate the site of the error by including the format string.

Reply:
Agreed, but do you think it is OK to ignore the keyword arguments with this approach? 
Probably?

Reply:
I tried a few versions of this. The one I like best inlcudes `lctx` and `fmt` but not `args`. Listing all of the args seems to be a lot of noise when there are probably several of them and they might be syntax objects (even noisier).


## Demo code 1

```
#lang racket
(require racket/syntax)
(format-id "~a-~a" 'mistake #'here)
```

## Short version of demo code 1

```
./racket/bin/racket qtest.rkt
format-id: contract violation
  expected: (or/c syntax? #f)
  given: "~a-~a"
  context...:
   body of "/home/racket/racket-src/racket/qtest.rkt"
```
## My preferred version of demo code 1

```
 ./racket/bin/racket qtest.rkt
format-id: contract violation
  expected: (or/c syntax? #f)
  given: "~a-~a"
  argument position: 1st
  other arguments...:
   'mistake
  context...:
   body of "/home/racket/racket-src/racket/qtest.rkt"
```

## Long version of demo code 1

```
$ ./racket/bin/racket qtest.rkt
format-id: contract violation
  expected: (or/c syntax? #f)
  given: "~a-~a"
  argument position: 1st
  other arguments...:
   'mistake
   #<syntax:qtest.rkt:3:30 here>
  context...:
   /home/racket/racket-src/racket/racket/collects/racket/syntax.rkt:109:0: format-id
   body of "/home/racket/racket-src/racket/qtest.rkt"
```

## Demo code 2

```
#lang racket
(require racket/syntax)
(format-id 'badness "~a-~a" 'mistake #'here)
```

## Short version:
```
 ./racket/bin/racket qtest.rkt
format-id: contract violation
  expected: (or/c syntax? #f)
  given: 'badness
  context...:
   body of "/home/racket/racket-src/racket/qtest.rkt"
```

## Long version

```
$ ./racket/bin/racket qtest.rkt
format-id: contract violation
  expected: (or/c syntax? #f)
  given: 'badness
  argument position: 1st
  other arguments...:
   "~a-~a"
   'mistake
   #<syntax:qtest.rkt:3:39 here>
  context...:
   /home/racket/racket-src/racket/racket/collects/racket/syntax.rkt:109:0: format-id
   body of "/home/racket/racket-src/racket/qtest.rkt"
```

Reply:
(let me know if this branch is too messy to merge)

Reply:
I think the benefit of including the args outweighs the costs. In particular, IMO including them is not really that much noise, and the extra information might help in some situations (sometimes, every bit of context helps). I think leaving out the keyword arguments is probably fine, but including only some of the positional arguments is potentially misleading to users.

We generally prefer to merge PRs by rebasing or squashing their commits rather than creating merge commits. To keep a PR branch up to date, you can rebase it with respect to origin/master and then force-push. (I'll volunteer to clean up the branch if you don't want to.)

Reply:
Ready to merge.

Reply:
This latest version lost the `apply` and the `fmt` argument.

Reply:
Fixed. Thanks for your patience.